### ISSUE TITLE
Generate hu/hv validation data for experiments 1–7

### ISSUE THREAD

**Issue by zeinali72** [zeinali72/swe-pinn#39]
## Summary

Currently, only experiment 8 has validation ground truth for all three output variables (`h`, `hu`, `hv`). Experiments 1–7 only validate against `h` (water depth). To enable consistent NSE/RMSE tracking across all variables and all experiments, we need to generate `hu` and `hv` validation data for experiments 1–7.

## Context

The training loop and Aim tracker now support tracking `nse_hu`, `rmse_hu`, `nse_hv`, `rmse_hv` for all experiments (see recent commit on `claude-cleanup`). However, experiments 1–7 lack the ground truth data to compute these metrics. This issue addresses generating that data.

### Experiment 1 — Analytical dam-break (flat domain)

**`hv = 0`** everywhere (1D problem in x-direction, no y-velocity).

**`hu`** has an analytical solution. The existing analytical solution for `h` is:
```python
h(x, t) = max(0, (7/3) * n² * u² * (u*t - x))^(3/7)
```
where `u = u_const` (constant velocity behind the wave front). Therefore:
```python
hu(x, t) = h(x, t) * u_const   # behind the wave front (where h > 0)
hu(x, t) = 0                     # ahead of the wave front (where h = 0)
```

**Action**: Add `hu_exact()` to `src/physics/analytical.py` and update experiment 1's `validation_fn` to compute `nse_hu`, `rmse_hu`, `nse_hv`, `rmse_hv`.

### Experiment 2 — Building obstacle

Validation data is loaded from `.npy` files generated from InfoWorks ICM simulation output.

**Action**: Regenerate the validation `.npy` file to include 6 columns `[t, x, y, h, u, v]` instead of just `[t, x, y, h]`. This requires reprocessing the original ICM CSV with the updated preprocessing pipeline (see #38). Then experiment 2's `validation_fn` already handles hu/hv when the data has 6 columns.

### Experiments 3–7 — Terrain slope / irregular domains

Same situation as experiment 2 — validation gauge files are generated from ICM simulation output.

**Action**: Regenerate each experiment's `validation_gauges.npy` (or `validation_gauges_ground_truth.npy`) to include all 6 columns `[t, x, y, h, u, v]`. The shared training loop already computes hu/hv metrics when `val_targets` has 3+ columns, so no code changes are needed once the data files are updated.

## Data File Inventory

| Experiment | Validation File | Current Columns | Target Columns |
|---|---|---|---|
| 1 | Analytical (generated in code) | `h` only | `h, hu, hv` (from analytical solution) |
| 2 | `validation_sample.npy` | `[t,x,y,h]` (4 cols) | `[t,x,y,h,u,v]` (6 cols) |
| 3 | `validation_gauges.npy` | needs audit | `[t,x,y,h,u,v]` (6 cols) |
| 4 | `validation_gauges_ground_truth.npy` | needs audit | `[t,x,y,h,u,v]` (6 cols) |
| 5 | `validation_gauges_ground_truth.npy` | needs audit | `[t,x,y,h,u,v]` (6 cols) |
| 6 | `validation_gauges_ground_truth.npy` | needs audit | `[t,x,y,h,u,v]` (6 cols) |
| 7 | `validation_gauges_ground_truth.npy` | needs audit | `[t,x,y,h,u,v]` (6 cols) |

## Acceptance Criteria

- [ ] `hu_exact()` added to `src/physics/analytical.py`
- [ ] Experiment 1 `validation_fn` reports `nse_hu`, `rmse_hu`, `nse_hv`, `rmse_hv`
- [ ] Experiments 2–7 validation `.npy` files regenerated with 6 columns `[t,x,y,h,u,v]`
- [ ] All experiments show `nse_hu`, `rmse_hu`, `nse_hv`, `rmse_hv` in Aim tracker when trained
- [ ] Consistent metric keys across all experiments in Aim

## Dependencies

- #38 (scripts cleanup) — the preprocessing pipeline must correctly convert speed+angle to u/v before regenerating validation files
